lib/response-cleaners/index.js

```diff
@@ -17,36 +13,33 @@ function arrayify(items) {
 // - @param {Object} response
 // - @return {Object} the modified response
 function ensureCollections(collectionPaths, response) {
-	var items, formats, listItems, basket;
-	_.each(collectionPaths, function checkLength(item) {
-		var parts = item.split('.');
-		var allPartsButLast = _.initial(parts);
-		var lastPart = _.last(parts);
-		var parents = _.reduce(allPartsButLast, function (chain, part) {
-			return chain.pluck(part).compact().flatten();
-		}, _([response])).value();
-
-		parents.map(function (parent) {
-			if (parent[lastPart]) {
-				parent[lastPart] = arrayify(parent[lastPart]);
-			} else {
-				parent[lastPart] = [];
-			}
-		});
-	});
-
-	basket = response.basket;
-	if (basket) {
-		if (basket.basketItems.basketItem) {
-			basket.basketItems = basket.basketItems.basketItem;
-		} else {
-			basket.basketItems = [];
+	_.each(collectionPaths, function ensureCollection(collectionPath) {
+		var secondLastItemPath = _.dropRight(collectionPath.path.split('.')).join('.');
+		var secondLastItem = _.get(response, secondLastItemPath);
+		var lastItem = _.get(response, collectionPath.path);
+		var destination = collectionPath.destination ? collectionPath.destination : collectionPath.path;
+		if (lastItem === undefined && secondLastItem === undefined) {
+			return;
 		}
-	}
+		if (lastItem === undefined && secondLastItem !== undefined) {
+			_.set(response, secondLastItemPath, []);
+			return;
+		}
+		if (typeof lastItem === 'string') {
+			_.set(response, destination, []);
```

Comment:
this is because when you get an xml response with an empty element that contains whitespace (specifically line breaks)

```
<someCollection>

</someCollection>
```

as opposed to:

```
<someCollection>

</someCollection>
```

the xml2js library gives you an object like `{ someCollection: '\n ' } ` which we should treat as a string. I don't think there are any cases where this would contain string data we'd actually want to keep.

If you're listing it as a collection to be flattened, surely you're not expecting string content.